aten/src/ATen/native/cuda/Sort.cpp

```diff
@@ -161,20 +111,11 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
   if (!indices_tmp->is_same(indices)) {
     indices.copy_(*indices_tmp);
   }
-  return std::forward_as_tuple(values, indices);
-}
-
-std::tuple<Tensor &,Tensor &> sort_out_cuda(const Tensor & self, int64_t dim, bool descending, Tensor & values, Tensor & indices) {
-  return sort_out_stable_cuda(self, /*stable=*/false, dim, descending, values, indices);
 }
 
-std::tuple<Tensor,Tensor> sort_stable_cuda(const Tensor & self, c10::optional<bool> stable, int64_t dim, bool descending) {
-  Tensor values, indices;
-  return sort_out_stable_cuda(self, stable, dim, descending, values, indices);
-}
-
-std::tuple<Tensor,Tensor> sort_cuda(const Tensor & self, int64_t dim, bool descending) {
-  return sort_stable_cuda(self, /*stable=*/false, dim, descending);
-}
+// TODO: we should handle this accordingly when we start using REGISTER_HIP_DISPATCH,
+// since REGISTER_DISPATCH won't work in this cpp file.
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+REGISTER_CUDA_DISPATCH(sort_stub, &sort_cuda_kernel);
```

Comment:
where did this come from?

Reply:
I don't think I got what you mean with this question. I needed this, since I am using `DispatchStub`, here.

Reply:
Just looked at this a little more. Is using `DispatchStub` for the cuda kernel actually necessary? My understanding of `DispatchStub` is:
- it allows you to do dynamic dispatch (technically redundant, since the dispatcher lets you do this too)
- it re-compiles the cpu kernel with multiple vector instruction sets

So the `sort` kernel is already using `DispatchStub` in the cpu kernel (which results in having to do 2 dynamic dispatches - one to go to CPU, and a second to go to the correct CPU vector instruction set).

But it looks like the cuda kernel in master today doesn't bother with dispatch stub. We use the dispatcher to dynamically dispatch to the cuda kernel and that's it. If the `REGISTER_CUDA_DISPATCH` is important for some other functionality (like hip support), then we can probably save that for a future PR, since it will also result in a second dynamic dispatch in the cuda path.

Lmk what you think!

Reply:
It's true that I didn't really need `DispatchStub`. That said, I think it's nice here since we get to unify common `CPU` and `CUDA` implementation, in exchange of a little overhead (I think). Do you think it's better to duplicate the code?

> it re-compiles the cpu kernel with multiple vector instruction sets

I thought that every source file inside `native/cpu` got compiled with different instructions sets (not really dispatched to them, though). Am I missing something?

Reply:
Ok, I think it's pretty reasonable (you're right, and we make that same tradeoff in a bunch of other ops that are implemented with DispatchStub). Sounds good!

> I thought that every source file inside native/cpu got compiled with different instructions sets (not really dispatched to them, though). Am I missing something?

Yep, that's my understanding too. And then the `sort_stub()` call is technically a dynamic dispatch that will pick the "correct" pre-compiled vectorized cpu kernel (in addition to potentially picking the cuda kernel), based on whatever hardware you happening to be using when running the op.